plus/provider/src/main/java/org/sakaiproject/plus/ProviderServlet.java

```diff
@@ -491,7 +491,13 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 			// Check if we are loop-backing on the same server, and already logged in as same user
 			Session sess = sessionManager.getCurrentSession();
 			String serverUrl = SakaiBLTIUtil.getOurServerUrl();
-			String ext_sakai_server = (String) payload.get("ext_sakai_server");
+			String iss = launch.tenant.getIssuer();
+			if ( StringUtils.equals(iss, serverUrl) ) {
+				log.debug("Running loopback id={} serverUrl={} iss={}", sess.getId(), serverUrl,iss);
+			} else {
+				sess.clear();
```

Comment:
if you have to clear this here it makes me think that it didn't get cleared where it should've...

Reply:
This was code that just never made it in.  If you look at the lines around it - it was preparing to clear the session, gathering data to make the choice and then it never did the actual clear().  There is not "two places" - there is "one place" and this is it and it was missing.

Reply:
I am sorry Dr Chuck if I was vague what I meant is that the RequestFilter should be doing this for you.

But it sounds like your handling a specific workflow so all is well, was just a thought.